### PR TITLE
Validate that publication date is in the past

### LINKED ISSUE #329: Add validation to publication date to make sure it is in the past.
While the year validation seems to be working, I can pick a month/day that are in the future.

related to #373

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #329

### CODE REVIEW

File: spec/validators/created_in_past_validator_spec.rb [hunk 1]
```diff
@@ -29,13 +29,41 @@
...
+      let(:today) { Time.zone.today }
 
-      it 'has errors' do
-        expect(record.errors.full_messages).to eq ['Created edtf must be in the past']
+      context 'with a date after the current year' do
+        let(:year) { today.year + 1 }
+        let(:value) { EDTF.parse(year.to_s) }
+
+        it 'has errors' do
+          expect(record.errors.full_messages).to eq ['Created edtf must be in the past']
```
Comment (justinlittman) [inline spec/validators/created_in_past_validator_spec.rb:41]: Is this a useful error message?

File: app/validators/created_in_past_validator.rb [hunk 2]
```diff
@@ -20,6 +20,6 @@ def validate_each(record, attribute, value)
   def validate_date(record, attribute, value, prefix = nil)
     prefix &&= "#{prefix} "
     record.errors.add(attribute, "#{prefix}must have a four digit year") if Settings.earliest_year > value.year
-    record.errors.add(attribute, "#{prefix}must be in the past") if Time.zone.today.year < value.year
+    record.errors.add(attribute, "#{prefix}must be in the past") if Time.zone.today < value
```
Comment (mjgiarlo) [inline app/validators/created_in_past_validator.rb:23]: What is `value` here? Is it a string, a date, an edtf, etc.?

File: spec/validators/created_in_past_validator_spec.rb (hunk 1 above)
Comment (mjgiarlo) [reply spec/validators/created_in_past_validator_spec.rb:41]: For a developer, sure. For an end-user, not so much.

Comment (jcoyne) [reply spec/validators/created_in_past_validator_spec.rb:41]: The end user will never look at `full_messages`, so no problem.

File: app/validators/created_in_past_validator.rb (hunk 2 above)
Comment (jcoyne) [reply app/validators/created_in_past_validator.rb:23]: It is a Date

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**